File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        numbers = list(set(range(1, n + 1)).difference(set(banned)))
        numbers.sort()
        count = 0
        sum = 0
        for n in numbers:
            sum += n
            if sum > maxSum:
                return count
            count += 1
        return count

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.rs
use std::collections::HashSet;

impl Solution {
    pub fn max_count(banned: Vec<i32>, n: i32, max_sum: i32) -> i32 {
        let range = HashSet::<i32>::from_iter((1..=n)
                            .collect::<Vec<_>>()
                            .into_iter());
        let banned = HashSet::from_iter(banned.into_iter());
        let mut numbers = range.difference(&banned)
                               .into_iter().collect::<Vec<_>>();
        numbers.sort();
        let mut sum = 0;
        for (i, n) in numbers.iter().enumerate() {
            sum += *n;
            if sum > max_sum {
                return i as i32;
            }
        }
        return numbers.len() as i32;
    }
}